streams/src/main/scala/io/confluent/examples/streams/MapFunctionScalaExample.scala

```diff
@@ -0,0 +1,73 @@
+package io.confluent.examples.streams
+
+import java.util.Properties
+
+import io.confluent.examples.streams.utils.SystemTimestampExtractor
+import org.apache.kafka.common.serialization.{ByteArrayDeserializer, ByteArraySerializer, StringDeserializer, StringSerializer}
+import org.apache.kafka.streams.kstream.{KStream, KStreamBuilder, KeyValue}
+import org.apache.kafka.streams.{KafkaStreaming, StreamingConfig}
+
+/**
+  * Demonstrates how to perform simple, state-less transformations via map functions.
+  *
+  * Use cases include e.g. basic data sanitization, data anonymization by obfuscating sensitive data
+  * fields (such as personally identifiable information aka PII).
+  *
+  * Requires Scala 2.12 for Java 8 and Java lambda support.
```

Comment:
Two things:

* Java 8 is supported since Scala 2.10.4
* SAM (single abstract method) support is available in 2.11.x via -Xexperimental